server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3483,6 +3483,8 @@ private Ternary<List<Long>, Integer, String[]> searchForDiskOfferingsIdsAndCount
                     diskOfferingSearch.and("displayOffering", diskOfferingSearch.entity().getDisplayOffering(), Op.EQ);
                 }
 
+                diskOfferingSearch.and("activeState", diskOfferingSearch.entity().getState(), Op.EQ);
```

Comment:
wouldn't the L3465 be the same as this? Should we change the searchCriteria to use state rather than activeState?

Reply:
I agree. @weizhouapache I have added a suggestion on what it should be. The default value of state when not set by user is Active.

Reply:
L3465 is executed only when `state != null`
if state is not passed in the API parameter, it will not be executed.

Reply:
@weizhouapache the state is never null. It was not working at all earlier because we were setting `activeState` which hasn't been defined in the search builder.

Below is the getState method from listDiskOfferingsCmd. 
```
    public State getState() {
        if (StringUtils.isBlank(diskOfferingState)) {
            return Active;
        }
        State state = EnumUtils.getEnumIgnoreCase(State.class, diskOfferingState);
        if (!diskOfferingState.equalsIgnoreCase("all") && state == null) {
            throw new IllegalArgumentException("Invalid state value: " + diskOfferingState);
        }
        return state;
    }

```